src/main/java/se/bjurr/violations/lib/parsers/ViolationParserUtils.java

```diff
@@ -20,6 +23,46 @@
 import se.bjurr.violations.lib.util.Optional;
 
 public final class ViolationParserUtils {
+  private static final Map<String, Character> XML_ESCAPE_CHARACTER_MAP;
+  private static final String[] XML_ESCAPE_CHARACTERS;
+  private static final char XML_ESCAPE_START = '&';
+
+  static {
+    Map<String, Character> temp = new HashMap<>();
+    temp.put("&apos;", '\'');
+    temp.put("&quot;", '\"');
+    temp.put("&amp;", '&');
+    temp.put("&lt;", '<');
+    temp.put("&gt;", '>');
+    XML_ESCAPE_CHARACTER_MAP = Collections.unmodifiableMap(temp);
+    XML_ESCAPE_CHARACTERS = temp.keySet().toArray(new String[0]);
+  }
+
+  private static String unEscapeXml(String input) {
+    StringBuilder result = new StringBuilder(input.length());
+    for (int i = 0; i < input.length(); ) {
+      char current = input.charAt(i);
+
+      boolean isValidXmlEscape = false;
+      if (current == XML_ESCAPE_START) {
+        for (String escapeCharacter : XML_ESCAPE_CHARACTERS) {
+          if (input.startsWith(escapeCharacter, i)) {
+            result.append(XML_ESCAPE_CHARACTER_MAP.get(escapeCharacter));
+            i += escapeCharacter.length();
+            isValidXmlEscape = true;
+            break;
+          }
+        }
+      }
+
+      if (!isValidXmlEscape) {
+        result.append(current);
+        i++;
+      }
+    }
+    return result.toString();
+  }
+
   public static String asString(XMLStreamReader xmlr) throws Exception {
```

Comment:
I think this method is probably unescaped from the XML library already? No new tests passed or failed when I unescaped the return value here so I left it unmodified.